lib/Parser/Parse.cpp

```diff
@@ -5094,14 +5102,7 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
                 Scope* paramScope = pnodeFnc->sxFnc.pnodeScopes->sxBlock.scope;
                 Assert(paramScope != nullptr);
 
-                if (pnodeFnc->sxFnc.CallsEval() || pnodeFnc->sxFnc.ChildCallsEval())
-                {
-                    if (!m_scriptContext->GetConfig()->IsES6DefaultArgsSplitScopeEnabled())
-                    {
-                        Error(ERREvalNotSupportedInParamScope);
-                    }
-                }
-                else
+                if (paramScope->GetCanMergeWithBodyScope())
```

Comment:
I am keeping it there for the eval case when I implement it later. If param scope has eval we will mark it as cannot merge from there.